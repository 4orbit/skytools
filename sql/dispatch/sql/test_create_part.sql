

\set ECHO none
set log_error_verbosity = 'terse';
set client_min_messages = 'warning';

\i create_partition.sql
\set ECHO all

create table events (
    id int4 primary key,
    txt text not null,
    ctime timestamptz not null default now(),
    someval int4 check (someval > 0)
);
create index ctime_idx on events (ctime);

select create_partition('events', 'events_2011_01', 'id', 'ctime', '2011-01-01', 'month');
select create_partition('events', 'events_2011_01', 'id', 'ctime', '2011-01-01'::timestamptz, 'month');

select create_partition('events', 'events_2011_01', 'id', 'ctime', '2011-01-01'::timestamp, 'month');

select count(*) from pg_indexes where schemaname='public' and tablename = 'events_2011_01';
select count(*) from pg_constraint where conrelid = 'public.events_2011_01'::regclass;

-- \d events_2011_01

