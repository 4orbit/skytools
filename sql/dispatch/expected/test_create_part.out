\set ECHO none
create table events (
    id int4 primary key,
    txt text not null,
    ctime timestamptz not null default now(),
    someval int4 check (someval > 0)
);
create index ctime_idx on events (ctime);
select create_partition('events', 'events_2011_01', 'id', 'ctime', '2011-01-01', 'month');
 create_partition 
------------------
                1
(1 row)

select create_partition('events', 'events_2011_01', 'id', 'ctime', '2011-01-01'::timestamptz, 'month');
 create_partition 
------------------
                0
(1 row)

select create_partition('events', 'events_2011_01', 'id', 'ctime', '2011-01-01'::timestamp, 'month');
 create_partition 
------------------
                0
(1 row)

select count(*) from pg_indexes where schemaname='public' and tablename = 'events_2011_01';
 count 
-------
     2
(1 row)

select count(*) from pg_constraint where conrelid = 'public.events_2011_01'::regclass;
 count 
-------
     3
(1 row)

-- \d events_2011_01
